angular2/src/angular2TextMask.ts

```diff
@@ -36,19 +37,11 @@ export class MaskedInputDirective implements ControlValueAccessor {
 
   constructor(private renderer: Renderer, private element: ElementRef) {}
 
-  private setupMask() {
-    if (this.element.nativeElement.tagName === 'INPUT') {
-      // `textMask` directive is used directly on an input element
-      this.inputElement = this.element.nativeElement
-    } else {
-      // `textMask` directive is used on an abstracted input element, `ion-input`, `md-input`, etc
-      this.inputElement = this.element.nativeElement.getElementsByTagName('INPUT')[0]
-    }
-
-    if (this.inputElement) {
-      this.textMaskInputElement = createTextMaskInputElement(
-          Object.assign({inputElement: this.inputElement}, this.textMaskConfig)
-      )
+  ngOnChanges() {
+    this.setupMask()
+    if (this.textMaskInputElement !== undefined) {
+      this.textMaskInputElement.update(this.inputElement.value)
+      this._onChange(this.inputElement.value)
```

Comment:
Personally, I don't think we want that.. and it doesn't seem to do that in the example app..

I'm considering removing that line